### ISSUE TITLE
Resolver with cursor pagination arguments broken in 11.3.2

### ISSUE THREAD

**Issue by tobias-tengler** [ChilliCream/hotchocolate#3982]
**Describe the bug**
Trying to access pagination arguments in an Annotation-based resolver throws an exception, complaining that an item with this key has already been added to the `FieldCollection`.

The [`CustomCursorHandlerTests`](https://github.com/ChilliCream/hotchocolate/blob/main/src/HotChocolate/Core/test/Types.CursorPagination.Tests/CustomCursorHandlerTests.cs) fail as expected, so I'm not really sure how these issues slipped into the release...

**To Reproduce**
Try to access one of the pagination arguments:
```csharp
public class Query
{
    [UsePaging]
    public IEnumerable<User> GetUsers(int first)
    {
        return  new List<User>()
        {
            new User { Id = "123", Name = "123n" }, 
            new User { Id = "456", Name = "456n" }
        };
    }
}
```

Produces the following exception:

> System.ArgumentException: An item with the same key has already been added. Key: first
>          at System.Collections.Generic.Dictionary`2.TryInsert(TKey key, TValue value, InsertionBehavior behavior)
>          at HotChocolate.Types.FieldCollection`1..ctor(IEnumerable`1 fields, Boolean sortByName)
>          at HotChocolate.Types.FieldCollection`1.From(IEnumerable`1 fields, Boolean sortByName)
>          at HotChocolate.Types.OutputFieldBase`1..ctor(TDefinition definition, FieldCoordinate fieldCoordinate, Boolean sortArgumentsByName)
>          at HotChocolate.Types.ObjectField..ctor(ObjectFieldDefinition definition, FieldCoordinate fieldCoordinate, Boolean sortArgumentsByName)
>          at HotChocolate.Types.ObjectType.<>c__DisplayClass32_0.<OnCompleteType>b__1(ObjectFieldDefinition t)
>          at System.Linq.Enumerable.WhereSelectEnumerableIterator`2.ToList()
>          at System.Linq.Enumerable.ToList[TSource](IEnumerable`1 source)
>          at HotChocolate.Types.ObjectType.OnCompleteType(ITypeCompletionContext context, ObjectTypeDefinition definition)
>          at HotChocolate.Types.TypeSystemObjectBase`1.CompleteType(ITypeCompletionContext context)
>          at HotChocolate.Configuration.TypeInitializer.<CompleteTypes>g__CompleteType|31_0(RegisteredType registeredType)
>          at HotChocolate.Configuration.TypeInitializer.ProcessTypes(TypeDependencyKind kind, Func`2 action)
>          at HotChocolate.Configuration.TypeInitializer.CompleteTypes()
>          at HotChocolate.Configuration.TypeInitializer.Initialize(Func`1 schemaResolver, IReadOnlySchemaOptions options)
>          at HotChocolate.SchemaBuilder.Setup.InitializeTypes(SchemaBuilder builder, DescriptorContext context, IBindingLookup bindingLookup, IReadOnlyList`1 types, LazySchema lazySchema)
>          at HotChocolate.SchemaBuilder.Setup.Create(SchemaBuilder builder)
>          at HotChocolate.SchemaBuilder.Create()
>          at HotChocolate.SchemaBuilder.HotChocolate.ISchemaBuilder.Create()
>          at HotChocolate.Execution.RequestExecutorResolver.CreateSchemaAsync(NameString schemaName, RequestExecutorSetup options, IServiceProvider serviceProvider, CancellationToken cancellationToken)
>          at HotChocolate.Execution.RequestExecutorResolver.CreateSchemaServicesAsync(NameString schemaName, RequestExecutorSetup options, CancellationToken cancellationToken)
>          at HotChocolate.Execution.RequestExecutorResolver.GetRequestExecutorNoLockAsync(NameString schemaName, CancellationToken cancellationToken)
>          at HotChocolate.Execution.RequestExecutorResolver.GetRequestExecutorAsync(NameString schemaName, CancellationToken cancellationToken)
>          at HotChocolate.Execution.RequestExecutorProxy.GetRequestExecutorAsync(CancellationToken cancellationToken)
>          at HotChocolate.AspNetCore.HttpPostMiddleware.HandleRequestAsync(HttpContext context, AllowedContentType contentType)
>          at HotChocolate.AspNetCore.HttpPostMiddleware.InvokeAsync(HttpContext context)
>          at HotChocolate.AspNetCore.WebSocketSubscriptionMiddleware.InvokeAsync(HttpContext context)
>          at Microsoft.AspNetCore.Routing.EndpointMiddleware.<Invoke>g__AwaitRequestTask|6_0(Endpoint endpoint, Task requestTask, ILogger logger)
>          at Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Http.HttpProtocol.ProcessRequests[TContext](IHttpApplication`1 application)

**Expected behavior**
These parameters should be accessible like in 11.3.1 and no exception should be thrown.

**Desktop (please complete the following information):**
- Version 11.3.2

**Comment by michaelstaib** [comment]
This is the change in 11.3.2:
https://github.com/ChilliCream/hotchocolate/commit/84f25b8d68987d450c619d2c156d0b26bd34ce9c

**Comment by michaelstaib** [comment]
This should not have affected this bit.

**Comment by tobias-tengler** [comment]
@michaelstaib a rather old PR adding a `AllowBackwardsPagination` flag was also merged onto main and released with v11.3.2 - I think that's the culprit.

**Comment by michaelstaib** [comment]
@PascalSenn can you have a look ... this one actually slipped in by accident and was slated for 11.4. It now causes an issue with the arguments. @tobias-tengler good catch!

**Comment by michaelmccord** [comment]
@michaelstaib will the fix from @tobias-tengler not be merged back into the 11.3.x line?

**Comment by tobias-tengler** [comment]
@michaelmccord Michael did the fix and I'm not 100% sure, whether the PR can be a patch release. An upgrade would cause a change in the schema, since now there are new comments, default values and a `nodes` field. Semantically it can't be patch release.
@michaelstaib Maybe we could extract a patch for the issue from your PR onto main and release it as 11.3.3? I think it's definitely needed, since this is the latest v11 version, many people will likely stay on.

**Comment by michaelstaib** [comment]
Sure, why not.

**Comment by michaelstaib** [comment]
But it should be 11.4

**Comment by michaelmccord** [comment]
@michaelstaib it would be much appreciated, there are many changes in the 12.x line that I want more time to explore before integrating them into my product.

**Comment by michaelmccord** [comment]
@michaelstaib @tobias-tengler should I file a new issue for this or what's the procedure?

**Comment by tobias-tengler** [comment]
@michaelmccord A fix for this has been released as 11.3.3 🎉

**Comment by michaelmccord** [comment]
@tobias-tengler rock on! Thanks!
